vendor/engines/secure_rooms/db/migrate/20170316090613_add_unique_index_to_card_readers.rb

```diff
@@ -0,0 +1,12 @@
+class AddUniqueIndexToCardReaders < ActiveRecord::Migration
+
+  def change
+    add_index(
+      :secure_rooms_card_readers,
+      [:product_id, :card_reader_number, :control_device_number],
```

Comment:
When we're getting the data from the reader, we're just getting a controller ID and a reader ID, right? So if we allowed duplicates across products, we could be talking to the wrong product, right? I think globally unique is the way to go.